File: word/egzaminų užduotys/2014-IT-bandomasis/Skaicius.cpp
// EgzaminoUzduotis
#include <iostream>
#include <fstream>
#include <iomanip>
using namespace std;

const char RezF[]  = "U2rez.txt";
typedef int skaiciaiMas[1001];   

//----------------------------------------------------------
void GaukSkaicius(skaiciaiMas visi, int& dydis);
void Apversk(skaiciaiMas rez);
void RasykRezultata();
//----------------------------------------------------------
int main()
{
int i, kiek=0, num;
skaiciaiMas  duomenys, rezultatai;
for (i = 0; i<rand() % 999+10;i++)
{
    GaukSkaicius(duomenys, kiek);
}   
Apversk(rezultatai);
RasykRezultata();
  return 0;
}
//-------------------------------------------------------
void Apversk(skaiciaiMas rez)
{

}

void RasykRezultata()
{

}
void GaukSkaicius(skaiciaiMas visi, int& dydis) // visada kuria skirtingus skaièius 
{
     int i, j, sk, m;
     bool nera;
 do
 {
    nera=true;
    sk=rand() % 9999 + 1; // funkcija rand generuoja atsitiktinius skaièius
    i = 0;
     while (i < dydis && nera) {
        
            i++;
            if (visi[i] == sk) nera = false;
            }
 }
 while (!nera);
 dydis++;
 visi[dydis] = sk;
}

File: word/egzaminų užduotys/2014-IT-bandomasis/Skaicius.pas
program EgzaminoUzduotis;
const
  RezF  = 'U2rez.txt';  { rezultatø failo pavadinimas }
  
type skaiciaiMas = array[1..1000] of integer;

//***********************

procedure GaukSkaicius(var visi: skaiciaiMas; var dydis:integer); { visada kuria skirtingus skaièius }
var i, j, sk, m : integer;
 nera : boolean;
begin
 repeat
    nera:=true;
    sk:=random(9999)+1; { funkcija random gràþina skaièiø nuo 0 iki 9999 }
    i := 0;
     while (i < dydis) and (nera) do
        begin
            i:=i+1;
            if (visi[i] = sk) then nera := false;
        end;
 until nera;
 dydis := dydis + 1;
 visi[dydis] := sk;
end;

//***********************

function Apversk():skaiciaiMas;

begin

end;

//***********************

procedure RasykRezultata();

begin

end;

//***********************

var       i, kiek, num : integer;
  duomenys, rezultatai : skaiciaiMas;
begin
 randomize; { procedûra, kuri inicijuoja atsitiktiniø skaièiø generatoriø }
 kiek := 0;
 for i := 1 to random(900)+10 do
    GaukSkaicius(duomenys, kiek);
    
 rezultatai := Apversk();
 RasykRezultata();
end.
